eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/MempoolTxMonitor.scala

```diff
@@ -0,0 +1,194 @@
+          sendResult(replyTo, TxConfirmed, Some(messageAdapter))
+        } else {
+          Behaviors.same
+        }
+      case TxNotFound =>
+        log.warn("txid={} has been evicted from the mempool", tx.txid)
+        checkInputStatus(input)
+        Behaviors.same
+      case GetTxConfirmationsFailed(reason) =>
+        log.error("could not get tx confirmations", reason)
+        // We will retry when the next block is found.
+        Behaviors.same
+      case status: InputStatus =>
+        if (status.spentConfirmed) {
+          log.info("tx was evicted from the mempool: a conflicting transaction has been confirmed")
+          sendResult(replyTo, TxRejected(TxPublisher.ConflictingTxConfirmed))
+        } else if (status.spentUnconfirmed) {
+          log.info("tx was evicted from the mempool: a conflicting transaction replaced it")
+          sendResult(replyTo, TxRejected(TxPublisher.ConflictingTxUnconfirmed))
+        } else {
+          log.info("tx was evicted from the mempool: one of our wallet inputs disappeared")
+          sendResult(replyTo, TxRejected(TxPublisher.WalletInputGone))
+        }
+      case CheckInputFailed(reason) =>
+        log.error("could not check input status", reason)
+        sendResult(replyTo, TxRejected(TxPublisher.TxSkipped(retryNextBlock = true)), Some(messageAdapter))
+      case Stop =>
+        context.system.eventStream ! EventStream.Unsubscribe(messageAdapter)
+        Behaviors.stopped
+    }
+  }
+
+  def sendResult(replyTo: ActorRef[TxResult], result: TxResult, blockSubscriber_opt: Option[ActorRef[CurrentBlockCount]] = None): Behavior[Command] = {
+    blockSubscriber_opt.foreach(actor => context.system.eventStream ! EventStream.Unsubscribe(actor))
+    replyTo ! result
+    Behaviors.stopped
+  }
+
+  private def checkInputStatus(input: OutPoint): Unit = {
+    val checkInputTask = for {
+      parentConfirmations <- bitcoinClient.getTxConfirmations(input.txid)
+      spendableMempoolExcluded <- bitcoinClient.isTransactionOutputSpendable(input.txid, input.index.toInt, includeMempool = false)
+      spendableMempoolIncluded <- bitcoinClient.isTransactionOutputSpendable(input.txid, input.index.toInt, includeMempool = true)
+    } yield computeInputStatus(parentConfirmations, spendableMempoolExcluded, spendableMempoolIncluded)
+    context.pipeToSelf(checkInputTask) {
+      case Success(status) => status
+      case Failure(reason) => CheckInputFailed(reason)
+    }
+  }
+
+  private def computeInputStatus(parentConfirmations: Option[Int], spendableMempoolExcluded: Boolean, spendableMempoolIncluded: Boolean): InputStatus = {
+    parentConfirmations match {
+      case Some(0) => InputStatus(spentConfirmed = false, spentUnconfirmed = !spendableMempoolIncluded)
+      case Some(_) => InputStatus(!spendableMempoolExcluded, !spendableMempoolIncluded)
+      case None => InputStatus(spentConfirmed = false, spentUnconfirmed = false)
+    }
+  }
```

Comment:
That one is tricky. Not to self @pm47 : do a second pass

Reply:
If the spending tx is confirmed, we want `InputStatus(spentConfirmed = true, spentUnconfirmed = false)` right? Then I suppose it should be:

```suggestion
  private def computeInputStatus(parentConfirmations: Option[Int], spendableMempoolExcluded: Boolean, spendableMempoolIncluded: Boolean): InputStatus = {
    parentConfirmations match {
      case Some(0) => InputStatus(spentConfirmed = false, spentUnconfirmed = !spendableMempoolIncluded)
      case Some(_) => InputStatus(!spendableMempoolExcluded, spendableMempoolExcluded && !spendableMempoolIncluded)
      case None => InputStatus(spentConfirmed = false, spentUnconfirmed = false)
    }
  }
```

Reply:
Also, I think you could completely remove the handler for `Some(0)`, it is redundant (even with my proposed modification).

Reply:
> If the spending tx is confirmed, we want InputStatus(spentConfirmed = true, spentUnconfirmed = false) right? 

You're right, that would make more sense. Done in https://github.com/ACINQ/eclair/pull/1844/commits/2ca2811f6fc503436e8e555f54094f52fc548fd4.

> Also, I think you could completely remove the handler for Some(0), it is redundant (even with my proposed modification).

Unfortunately we can't. In the case `Some(0)`, it means the parent is unconfirmed, so `spendableMempoolExcluded` will always be `false` (that input cannot be spendable since it doesn't even exist when you exclude the mempool). So we'd have `spentConfirmed = true` which is incorrect.